test/system/docker_live_restore_test.go

```diff
@@ -0,0 +1,172 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build aws aws_edge baremetal packet
+// +build e2e
+
+package system_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	testutil "github.com/kinvolk/lokomotive/test/components/util"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+// Define manifests as YAML and then unmarshal it to a Go struct so that it is easier to write and
+// debug, as it can be copy-pasted to a file and applied manually.
+const dockerLiveRestoreOnNodesDSManifest = `apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  generateName: test-live-restore-
+spec:
+  selector:
+    matchLabels:
+      name: test-live-restore
+  template:
+    metadata:
+      labels:
+        name: test-live-restore
+    spec:
+      tolerations:
+      - key: ""
+        operator: Exists
+      terminationGracePeriodSeconds: 1
+      initContainers:
+      - name: test-live-restore
+        image: quay.io/kinvolk/docker:stable
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - 'docker info -f {{.LiveRestoreEnabled}} | grep true'
+        volumeMounts:
+        - name: docker-socket
+          mountPath: /var/run/docker.sock
+
+      containers:
+      - name: wait
+        image: quay.io/kinvolk/docker:stable
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - sleep infinity
+      volumes:
+      - name: docker-socket
+        hostPath:
+          path: /var/run/docker.sock
+`
+
+const dockerLogOptsOnNodesDSManifest = `apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  generateName: test-log-opts-
+spec:
+  selector:
+    matchLabels:
+      name: test-log-opts
+  template:
+    metadata:
+      labels:
+        name: test-log-opts
+    spec:
+      tolerations:
+      - key: ""
+        operator: Exists
+      terminationGracePeriodSeconds: 1
+      containers:
+      - name: wait
+        image: quay.io/kinvolk/jq:1.6
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - sleep infinity
+
+      initContainers:
+      - name: test-log-opts-1
+        image: quay.io/kinvolk/jq:1.6
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        args:
+        - -c
+        - jq '."log-opts"."max-size"' /etc/docker/daemon.json | grep 100m
+        volumeMounts:
+        - name: docker-config
+          mountPath: /etc/docker/daemon.json
+      - name: test-log-opts-2
+        image: quay.io/kinvolk/jq:1.6
```

Comment:
How is this image maintained? Maybe we can bind-mount `jq` from host?

Reply:
@surajssd what about this?

Reply:
Can't use this because it errors as:

```
jq: error while loading shared libraries: libjq.so.1: cannot open shared object file: No such file or directory
```

Reply:
And if we mount the library as well? :smile:

Reply:
Ah, but it depends also on other libs... Still, how is the image built/maintained?

Reply:
Right now I am building and pushing it. I haven't put the Dockerfile anywhere.

```dockerfile
FROM registry.fedoraproject.org/fedora:32

RUN dnf -y install --setopt=tsflags=nodocs jq && \
    dnf clean all
```

Reply:
Maybe we can put it as a comment or in commit message, so it's easier to find in the future?